spring-integration-core/src/main/java/org/springframework/integration/support/leader/LockRegistryLeaderInitiator.java

```diff
@@ -92,6 +87,17 @@
 	 */
 	private final Candidate candidate;
 
+	/**
+	 * Flag to denote whether the {@link ExecutorService} was provided via the setter and
+	 * thus should not be shutdown when {@link #destroy()} is called
+	 */
+	private boolean executorServiceExplicitlySet;
```

Comment:
> Given that the variable was renamed to `executorServiceExplicitlySet`, based on your feedback, it doesn't really convey any meaning as to its use, hence why I previously called it `shutdownExecutorService`.

The `shutdownExecutorService` is more misleading for me. That may mean that there is a special configurator for this property and it really isn't related to the `executorService`.

The `executorServiceExplicitlySet` explicitly tells us that it related to the action around the `executorService` property and cannot be configured separately.

That is really self-explanatory for me. Not sure what is your confuse. But I'm fine to leave the JavaDoc here.
Only the concern is to add a period in the end of the sentence.

Reply:
>The executorServiceExplicitlySet explicitly tells us that it related to the action around the executorService property and cannot be configured separately.

OK I take your point, but again I am used to self documenting code hence I went for a descriptive name.  

Not a problem, every day is a school day! 😄